lib/cocoapods/sources_manager.rb

```diff
@@ -208,11 +253,14 @@ def update(source_name = nil, show_output = false)
           sources =  git_sources
         end
 
+        changed_spec_paths = {}
         sources.each do |source|
           UI.section "Updating spec repo `#{source.name}`" do
             Dir.chdir(source.repo) do
               begin
+                prev_commit_hash = (git! %w(rev-parse HEAD)).strip
```

Comment:
@segiddins should we also move UI code there into source?

I don't think it's suitable because we would need config object inside source to handle UI things.

I'm thinking of making a method returning a tuple variable including git pull output and changed paths. It would be sth like this:

```
[output, changed_spec_paths]  = source.update
```
Does this fit?